### ISSUE TITLE
docker system prune command crashes

### ISSUE THREAD

**Issue by egor-laufer** [docker/for-win#301]
### Expected behavior
`docker system prune` command works

### Actual behavior
`docker system prune` command crashes with `panic: runtime error: index out of range`

### Information

  - Diagnostic ID: C150D8A3-5644-4CAA-9ABD-F6EA04175BF3/2016-12-08_16-37-07
  - `Docker version 1.13.0-rc3, build 4d92237`
  - Windows 10 Pro, OS Build 14393.447

### Steps to reproduce the behavior

  1. run `docker system prune`
  2. stack trace is attached
[log.txt](https://github.com/docker/for-win/files/639546/log.txt)

`docker system prune -f` seem to work, suggesting that the problem happens when docker asks user for confirmation.

**Comment by thaJeztah** [comment]
ping @mlaventure PTAL

**Comment by mlaventure** [comment]
It looks like `docker/docker/pkg/term` doesn't support reading just 1 byte from a windows console atm :sweat_smile:

ping @jhowardmsft, it looks like the following should be added to `docker/docker/pkg/term/windows/ansi_reader.go`:

```diff
diff --git a/pkg/term/windows/ansi_reader.go b/pkg/term/windows/ansi_reader.go
index 0bccf878f..cb0b88356 100644
--- a/pkg/term/windows/ansi_reader.go
+++ b/pkg/term/windows/ansi_reader.go
@@ -115,6 +115,8 @@ func readInputEvents(fd uintptr, maxBytes int) ([]winterm.INPUT_RECORD, error) {
        countRecords := maxBytes / recordSize
        if countRecords > ansiterm.MAX_INPUT_EVENTS {
                countRecords = ansiterm.MAX_INPUT_EVENTS
+       } else if countRecords == 0 {
+               countRecords = 1
        }
        logger.Debugf("[windows] readInputEvents: Reading %v records (buffer size %v, record size %v)", countRecords, maxBytes, recordSize)
```

**Comment by jhowardmsft** [comment]
Hmmm. I'm not sure. The emulation in pkg/term/windows isn't used on a "native" Windows 10 console. @egor-laufer - what was the console you were using? cmd.exe? PowerShell? Or a third party console?

It doesn't repro using PS:
```
PS E:\docker\build> docker system prune
WARNING! This will remove:
        - all stopped containers
        - all volumes not used by at least one container
        - all networks not used by at least one container
        - all dangling images
Are you sure you want to continue? [y/N] y
Total reclaimed space: 0 B
PS E:\docker\build>
```


```
PS E:\docker\build> docker version
Client:
 Version:      1.14.0-dev
 API version:  1.26
 Go version:   go1.7.3
 Git commit:   3b31de6
 Built:        12/09/2016 17:24:30
 OS/Arch:      windows/amd64

Server:
 Version:      1.14.0-dev
 API version:  1.26 (minimum version 1.24)
 Go version:   go1.7.3
 Git commit:   3b31de6
 Built:        12/09/2016 17:24:30
 OS/Arch:      windows/amd64
 Experimental: false
```

**Comment by jhowardmsft** [comment]
Ah, it is used on a "legacy" console if you click this button in console properties:

![image](https://cloud.githubusercontent.com/assets/10522484/21058316/2ec732f8-bdf2-11e6-87ab-96a5ec02432e.png)

I have a repro. Investigating.

**Comment by jhowardmsft** [comment]
@mlaventure Your fix does cure the panic and allows the command to work. But I'm seeing a new issue which is the "y/N" input isn't echoed to the console (or a CRLF). Investigating why. (Note it's my last day in the office for the year today, so if I can't figure the second part, I'll submit a PR to the engine for the first part of the fix anyway).

**Comment by jhowardmsft** [comment]
Fix submitted in https://github.com/docker/docker/pull/29287, and second-part fixed by https://github.com/docker/docker/pull/29288

**Comment by rneugeba** [comment]
@egor-laufer this seems to be fixed in  1.13.0-rc4 which is shipping with Beta34. I've tried it with legacy console and without and it doesn't seem to crash anymore. I'm closing this issue, but please re-open if it is still an issue for you on the latest beta
